google/resource_dataproc_cluster.go

```diff
@@ -310,6 +310,19 @@ func resourceDataprocCluster() *schema.Resource {
 								},
 							},
 						},
+						"encryption_config": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"kms_key_name": {
+										Type:     schema.TypeString,
+										Optional: true,
```

Comment:
Sorry I didn't catch on the first pass, but I don't think `kms_key_name` is optional if the `encryption_block` has been defined. 

In HCL it's possible to have a top level field marked as optional, but have nested fields required. This is the equivalent of saying "encryption_config doesn't have to be specified, but if it is then kms_key_name must be present".

Reply:
agree